operators_sample_website/src/main/java/com/generalbytes/batm/server/extensions/website/RestServiceWebsite.java

```diff
@@ -0,0 +1,199 @@
+        } catch (Throwable e) {
+            log.error("Error - status", e);
+        }
+        return this.responseExpectationFailed();
+    }
+
+    /**
+     * https://localhost:7743/extensions/website/terminals
+     *
+     * @param apiKey Morphis API key
+     * @return list of active terminals ( active = pingDelay < 5 min ).
+     */
+    @GET
+    @Path("/terminals")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Object terminals(@HeaderParam("x_api_key") String apiKey) {
+
+        try {
+            IApiAccess iApiAccess = this.checkSecurity(apiKey);
+            return getTerminalsByApiKey(iApiAccess);
+        } catch (NullApiKeyException e) {
+            return this.responseInvalidApiKey();
+        } catch (Throwable e) {
+            log.error("Error - terminals", e);
+        }
+        return this.responseExpectationFailed();
+    }
+
+    /**
+     * Method helps collect terminals with a same Morphis API Access in CAS
+     *
+     * @param iApiAccess - Morphis API key
+     * @return List<ITerminal>
+     */
+    private List<ITerminal> getTerminalsByApiKey(IApiAccess iApiAccess) {
+        Collection<String> terminals = iApiAccess.getTerminalSerialNumbers();
+        List<ITerminal> filteredTerminals = new ArrayList<>();
+        terminals.forEach(terminalSerial -> {
+            ITerminal terminal = SellExtensions.getExtensionContext().findTerminalBySerialNumber(terminalSerial);
+            if (isOnline(terminal)) {
+                filteredTerminals.add(terminal);
+            }
+        });
+        return filteredTerminals;
+    }
+
+    /**
+     * Method filters terminals with long ping delay
+     *
+     * @param terminal - Iterminal terminal
+     * @return boolean
+     */
+    private boolean isOnline(ITerminal terminal) {
+        long now = System.currentTimeMillis();
+        return terminal.getLastPingAt() != null
+            && (terminal.getLastPingAt().getTime() + pingDelay) > now;
+    }
+
+    /**
+     * @param apiKey - Morphis API key
+     * @return IApiAccess - Authenticated API key
+     */
+    private IApiAccess checkSecurity(String apiKey) throws NullApiKeyException {
+        IApiAccess iApiAccess = SellExtensions.getExtensionContext().getAPIAccessByKey(apiKey);
+        if (iApiAccess == null) {
+            throw new NullApiKeyException();
+        }
+        return iApiAccess;
+    }
+
+    private Response responseBadRequest(String paramName) {
+        return Response.status(HttpServletResponse.SC_BAD_REQUEST).entity("{ \"error\": \"Parameter " + paramName + " can't be null\"}").build();
+    }
+
+    private Response responseInvalidApiKey() {
+        return Response.status(HttpServletResponse.SC_UNAUTHORIZED).entity("{ \"error\": \"Invalid x-api-key\" }").build();
+    }
+
+    private Response responseExpectationFailed() {
+        return Response.status(HttpServletResponse.SC_EXPECTATION_FAILED).entity("{ \"error\": \"Expectation failed\" }").build();
```

Comment:
Please don't use HTTP 417 - it's related to header, see more at: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/417

To be honest it will be better if you just keep ISE 500 here or don't even bother responding after failure at all. Response will end up ISE 500 anyway